vision_evaluation/evaluators.py

```diff
@@ -1157,3 +1157,49 @@ def add_predictions(self, predictions, targets):
 
     def get_report(self, **kwargs):
         return {f'{self._get_id()}': (float(self.total_error)**(1 / self.p) / self.total_num) if self.total_num else 0.0}
+
+
+class ConfusionMatrixEvaluator(Evaluator):
+    """
+    Confusion matrix evaluator
+    """
+    def reset(self):
+        super(ConfusionMatrixEvaluator, self).reset()
+        self.all_targets = np.array([])
+        self.all_predictions = np.array([])
+
+    def _get_id(self):
+        return f'confusion_matrix'
+
+    def add_predictions(self, predictions, targets):
+        """ Evaluate a batch of predictions.
+        Args:
+            predictions: the model output numpy array. Shape (N,)
+            targets: the ground truth labels. Shape (N,)
+        """
+        assert predictions.shape == targets.shape
+        assert len(targets.shape) == 1
+
+        self.all_predictions = np.append(self.all_predictions, predictions, axis=0)
+        self.all_targets = np.append(self.all_targets, targets, axis=0)
+
+    def get_report(self, **kwargs):
+        """Get the calculated value of the metric
+        Arrgs:
+            labels: List of labels to index the matrix. Array-like of shape (n_classes)
+            normalize: Boolean flag. If True, it normalizes the returned confusion matrix is normalized by the sum of the rows.
+        """
+        assert "labels" in kwargs
+        labels = kwargs["labels"]
+        assert len(labels) > 0
+        assert set(self.all_predictions).issubset(set(labels))
+        assert set(self.all_targets).issubset(set(labels))
+        normalize = kwargs.get('normalize', False)
+        eps = 1e-5
+        out = dict()
+        confusion_matrix = sm.confusion_matrix(self.all_targets, self.all_predictions, labels=labels).astype('float')
```

Comment:
out of curiosity do we have some numbers about how slow this could be? I remember it is very slow when number of classes is high

Reply:
just tested this, for 1000 classes and 1,000,000 (prediction, target) pairs, the computation took 0.22 seconds